src/codegen.cpp

```diff
@@ -1833,6 +1833,7 @@ class jl_codectx_t {
     // local var info. globals are not in here.
     SmallVector<jl_varinfo_t, 0> slots;
     std::map<int, jl_varinfo_t> phic_slots;
+    std::map<int, std::pair<Value*, Value*> > scope_restore;
```

Comment:
I thought we were preferring DenseMap since it has better performance and `[]` behavior, or maybe that does still need `.at()`, but at least has that operator?

Reply:
I thought std::map was better for small maps, but I don't really care either way.

Reply:
For small maps we should probably use `SmallDenseMap`

Reply:
Looks like the gcc version that we have on CI has a false positive warning in SmallDenseMap. I really don't think it's worth micro-optimizing the datastructure here, so I'm just gonna switch this back to std::map